Adds Cyborg data to AI's status screen

GunHog opened the pull request:
Adds all useful data about a Cyborg's health to the AI's status screen.
- Includes Cyborg name, health, cell charge, module, area name, and stat data.
- Strictly read-only, the AI must use the RD console to lock, release, or detonate its cyborgs.
- Cyborgs slaved to an AI will have that AI's name displayed in its status screen.

EDIT: Station Time added as well, for both the AI and Cyborgs!
EDIT2: "Locked Down" status tracking removed.

I am open to suggestions on how much data to provide to or hold from the AI. I am also open to any suggestions on how much data to provide a cyborg about its master AI.

Note: <b>Emagged cyborgs do not disappear from the AI's status screen if they were synced before.</b> The RD computer uses a different var (can_control) to show or hide cyborgs from the AI. An AI watching only its status screen would not know, but a robust and alert AI could cross-reference the two systems and know there is a problem.  I am unsure on if I should also remove a borg from the AI's status screen or not. Suggestions?
- Cyborg status screen: 
  ![screenshot 2015-01-17 00 15 17](https://cloud.githubusercontent.com/assets/4955510/5787928/f2ca6a76-9ddd-11e4-81ee-0f9c4327923c.png)
- AI status screen: _(Outdated, Cyborgs will no longer appear as locked down.)_
  ![screenshot 2015-01-17 00 10 06](https://cloud.githubusercontent.com/assets/4955510/5787915/38bb18c4-9ddd-11e4-8662-301611535dc4.png)

Commit:
Adds Cyborg data to AI's status screen

Adds all useful data about a Cyborg's health to the AI's status screen.
- Includes Cyborg name, health, cell charge, module, area name, lock,
and stat data.
- Strictly read-only.
- Cyborgs slaved to an AI will have that AI's name displayed in its
status screen.

WJohn commented:
holy shit that is hideous on too many levels to even start

ike709 commented:
How is it hideous? I like it.

GunHog commented:
Suggestions for improvement are welcome. I wanted something that would not require a new verb or button.

palpatine213 commented:
Mind adding a station time display to the status screen as well? I find it really annoying that there is no easy way to check how long the round has gone on as AI.

WJohn commented:
@ike709 It's bright pink all over the top of the screen. This is INSANELY disgusting to look at. I'm fine with this status stuff being put in the tab on the right, but don't clog up the screen with fluff that doesn't do anything and also painfully blocks your view.

ike709 commented:
@WJohn That is his PC stats... This PR adds the stuff under the Status tab ingame

GunHog commented:
Oooooh, THAT is what you meant? No no, that is not part of this update, I just forgot to turn the GPU tracker off. The only thing being added is in the AI "Status" screen.

Commit:
Adds Station Time!

Commit:
Adds Station Time for borgies too

Miauw62 commented:
The biggest issue here is that this makes it easy to spot emagged cyborgs, where they could lay low before, now you're probably going to have to order them to fuck shit up instantly if there's an AI around.

... I just don't know. The AI is already very all-seeing, I'm not sure if we should make it even moreso.

Commit:
Removes tracking of lockdown status, fixes counter

A cyborg's lockdown status has been removed from the tracker.
Cyborgs are now properly removed from the AIs borg list if they are
destroyed.

GunHog commented:
At its current iteration, emagged borgs remain on the AI's status panel. An AI checking the RD console would indeed have proof that a borg was forcefully disconnected from it, as currently an emagged borg does not show up on the RD console (my update does not change this).

Does that change the current meta? If I leave it as is, then the AI would *not* know the emag status of a borg until it checks the RD console - same as before. The reasons an AI would have for checking would be the same: Someone calls a borg out as rogue, or a borg having suspicious behavior such as excessive silence or refusing to respond to the AI's orders.

There is not much that seems to bind a cyborg to its AI. The AI knows when it gains a cyborg, but can only check up on them using the RD console, or directly looking at it. I wanted to further emphasize the AI-Cyborg relationship, but without it compromising the option of emagging them. I do not feel this information would cause an issue with emagging, but I could indeed be wrong.

Note: I found that the "Locked Down" status flashes for a second or two during the emagging process. Sadly, this means that I must remove that data. Also, I fixed the borg counter, destroyed borgs were not properly getting removed from the list index.

Miauw62 commented on code/modules/mob/living/silicon/ai/ai.dm:
```diff
@@ -224,6 +224,14 @@ var/list/ai_list = list()
 
 		if(!stat)
 			stat(null, text("System integrity: [(health+100)/2]%"))
+			stat(null, "Station Time: [worldtime2text()]")
```
STEALTH CHANGES (╯°□°）╯︵ ┻━┻

(:^])

WJohn commented:
Oh alright then that's much better.
